Kotlin/lib/src/main/java/com/wolt/blurhashkt/BlurHashDecoder.kt

```diff
@@ -2,13 +2,21 @@ package com.wolt.blurhashkt
 
 import android.graphics.Bitmap
 import android.graphics.Color
-import kotlin.math.PI
+import java.util.*
 import kotlin.math.cos
 import kotlin.math.pow
 import kotlin.math.withSign
 
+// this is to optimize the number of calculations for "Math.cos()",
+// is is slow and for many images with same size it can be cached, improving performance.
+// the improvement can be noticed with images bigger than 80x80
+private const val USE_CACHE_FOR_MATH_COS = true
```

Comment:
I don't think this flag is needed.

Reply:
The cache will consume memory. I wanted to add a way to use it or not, sometimes the performance is good enough using one or two small images.
Maybe we can add it as another parameter in the `decode` method with a default value?

Reply:
Is it maybe possible to detect when (e.g. by image size) it will by faster and auto apply it so that a user does not have to care about it? Otherwise I would like a param for the method better than the constant.

Reply:
enabling it by size could work. I am adding a screen with real-time benchmark results. Once we play with it we can decide.